vue/components/organisms/configurator/configurator.vue

```diff
@@ -486,6 +360,177 @@ export const Configurator = {
+                type: this.configuratorType
+            });
+
+            this.configurator.bind("changed_frame", frame => {
+                // sets the frame changed flag only if there was a
+                // previous frame set and then updates the frame
+                // key to the new one (internal copy)
+                this.frameChanged = Boolean(this.frameData);
+                this.frameData = frame;
+
+                // updates the value of the single frame view
+                // variable as the new view may have several frames
+                // or just one and the variable value may change
+                this.singleFrameView =
+                    (this.configurator.frames[this.configurator.view] || 1) === 1;
+
+                // only the visible instance of this component
+                // should be sending events it's considered to
+                // be the main/master one
+                if (this.elementDisplayed) {
+                    this.$store.commit("currentFrame", frame);
+                    this.$bus.trigger("changed_frame", this.configurator, frame);
+                }
+            });
+
+            this.configurator.bind("loaded", () => {
+                // updates the current frame in view with the one currently set
+                // in the configurator (to be used for data visibility purposes)
+                const frame = `${this.configurator.view}-${this.configurator.position}`;
+                this.frameData = frame;
+
+                // determines if the current view is a single frame view, meaning that
+                // no rotation operation is present for the view (static view)
+                this.singleFrameView =
+                    (this.configurator.frames[this.configurator.view] || 1) === 1;
+
+                // updates the local loading variable to the false value, indicating
+                // that the loading operation has just finished
+                this.loading = false;
+
+                // triggers the resize operation, as some pending resize operation
+                // may have been pilling up during the loading operation, during which
+                // is not possible to trigger resize operations
+                this.resize(this.size, this.width, this.height);
+
+                // updates the current frame in the store, this information can be used
+                // by listener to update their internal state
+                this.$store.commit("currentFrame", frame);
+            });
+
+            this.configurator.bind("highlighted_part", part => {
+                this.$bus.trigger("highlighted_part", this.configurator, part);
+            });
+
+            this.configurator.bind("lowlighted", () => {
+                this.$bus.trigger("lowlighted", this.configurator);
+            });
+
+            this.configurator.bind("ready", options => {
+                if (options?.origin && options.origin.includes("configurator")) {

```

Comment:
> If you are using optional chaining you can do this options?.origin?.includes("configurator")
You can't do that when you're calling a method because it will throw

> Also, I think you should look for a more specific origin as in, use .startsWith() (it is also more efficient)
Will do